jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/internal/HTTP2Flusher.java

```diff
@@ -106,6 +108,8 @@ public boolean append(HTTP2Session.Entry entry)
         try (AutoLock ignored = lock.lock())
         {
             closed = terminated;
+            if (closed instanceof HpackException.SessionException)
+                closed = null;
```

Comment:
Niggle... it is strange to set then clear closed.  How about:

```suggestion
            // Comment about why terminated is ignored if it is a SessionException
            closed = (terminated instanceof HpackException.SessionException) ? null : terminated ;
```

Reply:
I was able to make the condition more precise now, but it is longer.
I like better the "assign the default case; then look at conditions for the special case", rather than doing it in a one-liner.